Core/Src/ds3231.c

```diff
@@ -0,0 +1,80 @@
+#include "ds3231.h"
+
+void DS3231_GetDate(DS3231_t* pDS)
+{
+  uint8_t buf[4] = {3};
+
+  I2C_Transmit(pDS->Adress, buf, 1, 100);
+  I2C_Receive(pDS->Adress, buf, 4, 100);
+
+  DS3231_Mem_Date_t* pDate = (DS3231_Mem_Date_t*)buf;
+
+  pDS->Date.Date = pDate->Date.Date.Date10 * 10 + pDate->Date.Date.Date;
+  pDS->Date.Month = pDate->Date.Month.Month10 * 10 + pDate->Date.Month.Month;
+  pDS->Date.Year = pDate->Date.Year.Year10 * 10 + pDate->Date.Year.Year;
+  pDS->Date.Day = pDate->Day;
+}
+
+void DS3231_GetTime(DS3231_t* pDS)
```

Comment:
Please, add checking for `pDS` not to be `NULL` in all public functions. You can use `assert` to check the input parameters

Reply:
I would remind you of about this remark. Please, check pointers in public functions in order not to be `NULL`